### PR TITLE
feat(ci): wire the static half of the MiniMax-Provider-Verifier into CI

### PR DESCRIPTION
**Pull request by POWERFULMOVES**
## What this slice does

Wires the static half of the MiniMax-Provider-Verifier conformance gate into CI. The verifier submodule (`Pmoves-MiniMax-Provider-Verifier/`) is the load-bearing test for adding any new provider to the Mavis model cascade. The full conformance run requires real API calls, so it stays the operator's manual step on the local node where the keys live in `env.shared`. The static half (6 checks, no API calls) is what runs on every PR.

Rebased on top of `feat/mcpcli-wireup` (#2612) to inherit the verifier doc + the minimax MCP wire-up from the prior slice. The 4090's commits on that branch (pin the MCP version per F-07, tighten the pin test) are also in the base.

## 11 commits, 3-stacked group

**Functional (1-4):**
1. `feat(gate): add pmoves/tools/provider_verifier_gate.py` — the 6-check static helper
2. `test(gate): add 22 tests for provider_verifier_gate` (6 checks × pass + fail + edge cases, plus CLI)
3. `feat(ci): add .github/workflows/provider-verifier.yml` — the CI gate, job name `verifier-gate`
4. `refactor(gate): take verifier_submodule as a parameter` (lesson from the first test run)

**Docs (5):**
5. `docs(verifier): update PROVIDER_VERIFIER_GATE.md` — Gate in CI is now wired
6. `docs(bootstrap): add provider-verifier to the Known Roads table` — cold-start discoverability

**Wire + tests + close (6-11):**
7. `feat(merge-gate): wire verifier-gate into merge-decision` + `test(workflow)` cross-workflow contract (15 + 3 = 18 glue tests)
8. `fix(workflow): add issues:write to static-gate job` (so the PR-comment step on FAIL has the scope it needs)
9. `test(workflow): +3 tests for skipped-vs-failure + permissions`
10. `test(install): pin the operator's full-conformance run path` (6 tests on `verify.py` CLI + requirements.txt + Python version + doc cross-ref)
11. `agnote + learnings`: AGNOTE `Mavis::PROVIDER-VERIFIER-CI-GATE-CLAIM-RELEASE::2026-08-19` + LEARNINGS file with 5 patterns

## Surfaces touched

- `pmoves/tools/provider_verifier_gate.py` (new, 385 lines): the helper
- `pmoves/tests/unit/test_provider_verifier_gate.py` (new, 349 lines): 22 unit tests
- `pmoves/tests/test_provider_verifier_workflow.py` (new, 392 lines): 18 workflow-glue tests
- `pmoves/tests/test_provider_verifier_install.py` (new, 177 lines): 6 operator-path tests
- `.github/workflows/provider-verifier.yml` (new, 147 lines): the CI gate workflow
- `.github/workflows/merge-gate.yml` (modified): merge-decision now includes `verifier-gate`
- `pmoves/docs/operations/PROVIDER_VERIFIER_GATE.md` (modified): Gate in CI section updated
- `.claude/BOOTSTRAP.md` (modified): Known Roads table has 2 new rows
- `pmoves/docs/AGENTS/AGNOTE4482PHI.t1.md` (modified): CLAIM+RELEASE row appended
- `provider_verifier_ci_gate_LEARNINGS.md` (new): 5 patterns captured

## Verification

- 58 tests pass + 1 skip (`test_verifier_help_lists_required_flags` skips cleanly when `megfile` isn't installed locally)
  - 22 unit (gate helper)
  - 18 workflow-glue (the YAML contract + cross-workflow + permissions)
  - 6 install-path (operator's full-conformance run)
  - 12 from `test_mcpcli_wireup.py` (carried over from the rebase)
  - 1 from `test_load_bootstrap.py` (carried over)
- All YAML / JSON validates
- The 6 static checks pass on the live submodule state

## What this slice does NOT do (intentional)

- The full conformance run via `verify.py` — operator's manual step on the local node, where the API keys live in `env.shared` (per the F-07 supply-chain note that forbids exposing keys in workflow files)
- A new model-cascade provider — the gate is the HOW; new providers are the WHEN
- The `archon.crawl.*` retire-vs-keep call (per #2582, not in this lane)
- The PMOVES.AI branch protection follow-up (the SKIPPED apply from 2026-08-15) — separate slice
- The other 3 follow-ups from the mcpcli slice (3 AGENTS.md adoption items, v2 HiRAG promotion, .claude/mcp.json HiRAG registration) — separate slices

## Three-body

- delivery: Mavis (11 commits + 58 tests + 4 docs)
- control: DARKXSIDE (the operator's local env is the canonical full-run path; untouched by this slice)
- memory: this trail + the LEARNINGS file

## CHIT trail

Unsigned-local. Sign-trail is the operator-step (B850 has the signed lane).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/merge-gate.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
         run: python pmoves/tools/hardening_ratchet.py
 
   merge-decision:
-    needs: [python-tests, docker-build-validation, hardening-validation]
+    needs: [python-tests, docker-build-validation, hardening-validation, verifier-gate]
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply .github/workflows/merge-gate.yml]: C:/Users/russe/AppData/Local/Temp/claude/C--Users-russe-OneDrive-Documents-GitHub-PMOVES-AI/eb22c84f-ec1b-4e4a-9b90-10cde196c80e/scratchpad/p1.md

File: .github/workflows/provider-verifier.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          python -m pip install -r Pmoves-MiniMax-Provider-Verifier/requirements.txt || \
+            echo "::warning::verifier runtime deps not installed; the static gate still runs"
+
+      - name: Run the static gate
+        id: gate
+        shell: bash
+        run: |
+          # 6 checks, no API calls. Output: JSON line on stdout.
+          set +e
+          OUTPUT=$(py pmoves/tools/provider_verifier_gate.py --json 2>&1)
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply .github/workflows/provider-verifier.yml]: C:/Users/russe/AppData/Local/Temp/claude/C--Users-russe-OneDrive-Documents-GitHub-PMOVES-AI/eb22c84f-ec1b-4e4a-9b90-10cde196c80e/scratchpad/p2.md

File: pmoves/tools/provider_verifier_gate.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    if not isinstance(entries, list):
+        return CheckResult(
+            name="example_keys_are_placeholders",
+            passed=False,
+            detail="provider.json.example is not a list",
+        )
+    leaked: List[str] = []
+    for i, entry in enumerate(entries):
+        key = entry.get("api_key", "")
+        if key and key != PLACEHOLDER_API_KEY:
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply pmoves/tools/provider_verifier_gate.py]: C:/Users/russe/AppData/Local/Temp/claude/C--Users-russe-OneDrive-Documents-GitHub-PMOVES-AI/eb22c84f-ec1b-4e4a-9b90-10cde196c80e/scratchpad/p3.md
